src/Uno.UI/UI/Xaml/Internal/CoreServices.cs

```diff
@@ -17,11 +20,96 @@ internal class CoreServices
 
 		private VisualTree? _mainVisualTree;
 
+#if UNO_HAS_ENHANCED_LIFECYCLE
+
+		private static int _isAdditionalFrameRequested;
+
+		public EventManager EventManager { get; private set; }
+#endif
+
 		public CoreServices()
 		{
 			ContentRootCoordinator = new ContentRootCoordinator(this);
+#if UNO_HAS_ENHANCED_LIFECYCLE
+			EventManager = EventManager.Create();
+#endif
+		}
+
+#if UNO_HAS_ENHANCED_LIFECYCLE
+		private static XamlRoot? GetXamlRoot()
+		{
+			if (CoreServices.Instance.ContentRootCoordinator.ContentRoots.Count > 0)
+			{
+				return CoreServices.Instance.ContentRootCoordinator.ContentRoots[0].XamlRoot;
+			}
+
+			if (CoreServices.Instance.MainVisualTree is { } mainVisualTree)
+			{
+				return mainVisualTree.XamlRoot;
+			}
+
+			return null;
+		}
+
+		internal static void RequestAdditionalFrame()
+		{
+			if (GetXamlRoot() is { Bounds: { Width: not 0, Height: not 0 } } &&
+				Interlocked.CompareExchange(ref _isAdditionalFrameRequested, 1, 0) == 0)
+			{
+				// This lambda is intentionally static. It shouldn't capture anything to avoid allocations.
+				NativeDispatcher.Main.Enqueue(static () => OnTick(), NativeDispatcherPriority.Normal);
+			}
 		}
 
+		private static void OnTick()
+		{
+			Interlocked.Exchange(ref _isAdditionalFrameRequested, 0);
```

Comment:
Note: as you are not using the return value, there is no need to use an `Interlocked` here :)